frontend/src/app/components/block-fees-subsidy-graph/block-fees-subsidy-graph.component.ts

```diff
@@ -0,0 +1,551 @@
+      const isFiat = params.name.includes('USD');
+      if (isFiat === this.showFiat) return;
+
+      const isActivation = params.selected[params.name];
+      if (isFiat === isActivation) {
+        this.showFiat = true;
+        this.chartInstance.dispatchAction({ type: 'legendUnSelect', name: 'Subsidy' });
+        this.chartInstance.dispatchAction({ type: 'legendUnSelect', name: 'Fees' });
+        this.chartInstance.dispatchAction({ type: 'legendSelect',   name: 'Subsidy (USD)' });
+        this.chartInstance.dispatchAction({ type: 'legendSelect',   name: 'Fees (USD)' });
+      } else {
+        this.showFiat = false;
+        this.chartInstance.dispatchAction({ type: 'legendSelect',   name: 'Subsidy' });
+        this.chartInstance.dispatchAction({ type: 'legendSelect',   name: 'Fees' });
+        this.chartInstance.dispatchAction({ type: 'legendUnSelect', name: 'Subsidy (USD)' });
+        this.chartInstance.dispatchAction({ type: 'legendUnSelect', name: 'Fees (USD)' });
+      }
+    });
+
+    this.chartInstance.on('datazoom', (params) => {
+      if (params.silent || this.isLoading) {
+        return;
+      }
+      this.updateZoom = true;
+    });
+
+    this.chartInstance.on('click', (e) => {
+      this.zone.run(() => {
+        if (['24h', '3d'].includes(this.zoomTimeSpan)) {
+          const url = new RelativeUrlPipe(this.stateService).transform(`/block/${e.name}`);
+          if (e.event.event.shiftKey || e.event.event.ctrlKey || e.event.event.metaKey) {
+            window.open(url);
+          } else {
+            this.router.navigate([url]);
+          }
+        }
+      });
+    });
+  }
+
+  @HostListener('document:mouseup', ['$event'])
+  onMouseUp(event: MouseEvent) {
+    if (this.updateZoom) {
+      this.onZoom();
+      this.updateZoom = false;
+    }
+  }
+
+  isMobile() {
+    return (window.innerWidth <= 767.98);
+  }
+
+  initSubsidies(): { [key: number]: number } {
+    let blockReward = 50 * 100_000_000;
+    const subsidies = {};
+    for (let i = 0; i <= 33; i++) {
+      subsidies[i] = blockReward;
+      blockReward = Math.floor(blockReward / 2);
+    }
+    return subsidies;
+  }
+
+  onZoom() {
+    const option = this.chartInstance.getOption();
+    const timestamps = option.xAxis[1].data;
+    const startTimestamp = timestamps[option.dataZoom[0].startValue];
+    const endTimestamp = timestamps[option.dataZoom[0].endValue];
+
+    // Avoid to fetch new data if not needed
+    if (option.dataZoom[0].end - option.dataZoom[0].start >= this.zoomSpan + 0.1
+       || this.getTimeRangeFromTimespan(Math.floor(startTimestamp / 1000), Math.floor(endTimestamp / 1000), false) as number === this.getTimeRange(this.zoomTimeSpan)
+      ) {
+      this.zoomSpan = option.dataZoom[0].end - option.dataZoom[0].start;
+      this.zoomTimeSpan = this.getTimeRangeFromTimespan(Math.floor(startTimestamp / 1000), Math.floor(endTimestamp / 1000), true) as string;
+      return;
+    }
```

Comment:
I suspect this logic could get us into trouble once we have a few different granularities of data all spliced together.

for example, after playing with the chart for a bit I ended up in this slightly confusing situation where each half of screen is on a different timescale:
<img width="1054" alt="Screenshot 2024-05-14 at 9 22 39 PM" src="https://github.com/mempool/mempool/assets/83316221/ad5c5f2f-ec96-4e9f-a0c7-ebb6a38b0f9b">

Reply:
This is because we don't reload data when the user zooms out: this allows to keep showing precise data on zoom out, but it's quite confusing as you showed.

Reply:
making a fresh API call on *every* change of zoom seems a little excessive, but it is probably the safest approach.

otherwise maybe it's possible to do something like:
 - cache the data in separate "layers" for each zoom level (1 min, 5 min, 30 min etc).
 - fetch any data that's missing from the target layer/zoom level within this datetime range.
 - replace the echart data series with the result of flattening the cached layers <= the target zoom level.
 
then you're only fetching data you don't have already, and never have to worry about random spans of 1 min data messing up the time axis on your 2hr view etc.

but tbh the extra complexity might not be worth the benefit 🤷

Reply:
I thought about doing something similar, and came up to the same conclusion as you 😅 but that's definitely the most optimal solution